File: smash-ranks-android/app/src/main/java/com/garpr/android/preferences/persistent/PersistentIntegerPreference.java
package com.garpr.android.preferences.persistent;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.garpr.android.preferences.KeyValueStore;

public class PersistentIntegerPreference extends BasePersistentPreference<Integer> {

    public PersistentIntegerPreference(@NonNull final String key,
            @Nullable final Integer defaultValue, @NonNull final KeyValueStore keyValueStore) {
        super(key, defaultValue, keyValueStore);
    }

    @Nullable
    @Override
    public Integer get() {
        if (hasValueInStore()) {
            // at this point, returning the fallback value is impossible
            return getKeyValueStore().getInteger(getKey(), 0);
        } else {
            return getDefaultValue();
        }
    }

    @Override
    protected void performSet(@NonNull final Integer newValue, final boolean notifyListeners) {
        getKeyValueStore().setInteger(getKey(), newValue);

        if (notifyListeners) {
            notifyListeners();
        }
    }

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/preferences/persistent/PersistentIntegerPreference.kt
package com.garpr.android.preferences.persistent

import com.garpr.android.preferences.KeyValueStore

class PersistentIntegerPreference(
        key: String,
        defaultValue: Int?,
        keyValueStore: KeyValueStore
) : BasePersistentPreference<Int>(
        key,
        defaultValue,
        keyValueStore
) {

    override fun get(): Int? {
        return if (hasValueInStore) {
            // at this point, returning the fallback value is impossible
            keyValueStore.getInteger(key, 0)
        } else {
            defaultValue
        }
    }

    override fun performSet(newValue: Int) {
        keyValueStore.setInteger(key, newValue)
        notifyListeners()
    }

}
